File: 백준/Silver/1764. 듣보잡/듣보잡.py
N, M = map(int, input().split())

known = set()
unknown = set()

for _ in range(N):
    known.add(input())

for _ in range(M):
    unknown.add(input())

result = sorted(known & unknown)
print(len(result))
for name in result:
    print(name)

File: 백준/Silver/1764. 듣보잡/듣보잡.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);

const knownSet = new Set();
const unknownSet = new Set();

for (let i = 1; i <= N; i++) {
    knownSet.add(input[i]);
}

for (let i = N + 1; i <= N + M; i++) {
    unknownSet.add(input[i]);
}

const result = [...knownSet].filter(name => unknownSet.has(name));
result.sort();

console.log(result.length);
console.log(result.join('\n'));